sdk/src/vm.rs

```diff
@@ -9,14 +10,31 @@ pub const NO_DATA_BLOCK_ID: u32 = 0;
 
 /// Abort execution.
 pub fn abort(code: u32, message: Option<&str>) -> ! {
+    exit(code, message, None)
+}
+
+/// Exit from current message execution, with the specified code and an optional message and data.
+pub fn exit(code: u32, message: Option<&str>, data: Option<RawBytes>) -> ! {
     unsafe {
         let (message, message_len) = if let Some(m) = message {
             (m.as_ptr(), m.len())
         } else {
             (ptr::null(), 0)
         };
 
-        sys::vm::abort(code, message, message_len as u32);
+        // Ideally we would write this double if as if let Some(bytes) = data && bytes.len() > 0
+        // but the compiler doexn't accept it.
+        let blk_id = if let Some(bytes) = data {
+            if bytes.len() > 0 {
+                sys::ipld::block_create(DAG_CBOR, bytes.as_ptr(), bytes.len() as u32).unwrap()
+            } else {
+                NO_DATA_BLOCK_ID
+            }
+        } else {
+            NO_DATA_BLOCK_ID
+        };
```

Comment:
FYI

```suggestion
        let blk_id = match data {
            Some(bytes) if !bytes.is_empty() => {
                sys::ipld::block_create(DAG_CBOR, bytes.as_ptr(), bytes.len() as u32).unwrap()
            },
            _ => NO_DATA_BLOCK_ID,
        };
```

Reply:
ah, if let doesn't work but there is always plan match; ok, good.
Too bad it will be written to not use the option :)

Reply:
Yep. I just wanted to make sure you've seen match guards.